Cython/Compiler/Nodes.py

```diff
@@ -2411,11 +2421,16 @@ def analyse_declarations(self, env):
 
         type.is_const_method = self.is_const_method
         type.is_static_method = self.is_static_method
+        property = False
+        if self.decorators:
+            for _node in self.decorators:
+                if _node.decorator.is_name and _node.decorator.name == 'property':
```

Comment:
The repetition of this looping code seems to scream for a helper function, e.g. one that searches for a specific decorator (and maybe applies some safety checks internally).

Reply:
Merged with decorator analysis at the beginning of the function. Did you mean that I should refactor that into a helper function?